src/tree_builder/interface.rs

```diff
@@ -61,6 +61,10 @@ pub trait TreeSink {
     /// Do two handles refer to the same node?
     fn same_node(&self, x: Self::Handle, y: Self::Handle) -> bool;
 
+    /// Are two handles present in the same tree
+    /// https://html.spec.whatwg.org/multipage/infrastructure.html#home-subtree
```

Comment:
As far as I understand there are two cases where it will be false:

1) Fragment Parsing case i.e ```elem.innerHTML =  <input type="text" />``` . In this case the parser won't call ```associate_form_owner``` even when one of ```elem```'s ancestor is a form element . But the form owner will be set when the DocumentFragment's children are inserted into the context element, after parsing.

2) An inline script element removes the form element from the Document before the ```<input>``` is parsed and ```form_elem``` points to a form element that is not an ancestor of the input tag (the wierd table case)

I must admit that I don't fully understand case 1. In particular, why does the fragment parser algorithm initialize the parser's form element pointer to the closest form element ancestor of the context element? (https://html.spec.whatwg.org/multipage/syntax.html#parsing-html-fragments:form-element-pointer step 11)

I'm not sure if there are other cases that I might have missed.